### PR TITLE
[galaxie greek mnemonic] Add rules mitigating NFD characters

### PR DESCRIPTION
**Pull request by MakaraSok**
Rules mitigating NFD for characters with:
- single diacritic
- double diacritic
- triple diacritic

Version number and year of copyrights bumped.

Fixing: https://github.com/keymanapp/keyboards/issues/1264

### CODE REVIEW

**Comment by MakaraSok** [comment]
@AndreasJoswig Fixes have been made, but there is an issue with some characters which are, for some reason, not visible in FLEx field. It however is visible in the preview panel in the highlighted circle (see the screenshot below). The pattern is that this behavior usually happens to characters with `macron`,  `breve`, a combination of `diaeresis` and `circumflex accent`, and when `iota subscript` is present in characters with multiple diacritics.
image


Now, could you please test the fixes and see if you encounter the known issue above?

Here is the [keyboard package](https://drive.google.com/file/d/1XzGB0-maz0y4gsCInb6eCJ9IkV0kLuwR/view?usp=sharing) for testing.

**Comment by MakaraSok** [comment]
@AndreasJoswig Thank you for the prompt response. It is great that you didn't encounter any issue. Could you let me know what your FLEx and Keyman version are?

**Comment by AndreasJoswig** [comment]
@MakaraSok It is FLEx 9.0.9.0 RC under Windows 10, and Keyman 13.0.109.0

**Comment by LornaSIL** [comment]
If you've turned this into an NFC keyboard I'd rather the comment in HISTORY.md clearly say that. Also, the readme.htm should say it is an NFC keyboard and the description in the keyboard_info should say that.

Also, it seems significant enough that the version should be 3.3 (not 3.2.2).

**Comment by mcdurdin** [comment]
> If you've turned this into an NFC keyboard I'd rather the comment in HISTORY.md clearly say that. Also, the readme.htm should say it is an NFC keyboard and the description in the keyboard_info should say that.

The keyboard is still generating NFC -- this hasn't changed. This PR is making the keyboard more compatible with FLEx by adding rules to match NFD context.

File: release/g/galaxie_greek_mnemonic/source/help/galaxie_greek_mnemonic.php [hunk 1]
```diff
@@ -2213,6 +2213,7 @@
             </p>
 
                 <p>
+					Version 3.2.2 released 6 Jul 2020. Add rules to mitigate NFD issues.<br/>
```
Comment (mcdurdin) [inline release/g/galaxie_greek_mnemonic/source/help/galaxie_greek_mnemonic.php:2216]:
```suggestion
                	Version 3.3 released 6 Jul 2020. Add rules to mitigate NFD contextual matches and improve compatibility with FLEx.<br/>
```

Comment (MakaraSok) [reply release/g/galaxie_greek_mnemonic/source/help/galaxie_greek_mnemonic.php:2216]: Changed will be added in the next commit.

**Comment by MakaraSok** [comment]
> So, I repeat, I can recreate the success reported by Makara, but I cannot recreate the problem reported by him. All is well.

I tested this again in FLEx 9.0.9 using Charis SIL font with Graphite enabled and everything is OK. No error as you pointed out. :)

**Comment by LornaSIL** [comment]
@mcdurdin were you going to check this or do I need to?

**Comment by mcdurdin** [comment]
> @mcdurdin were you going to check this or do I need to?

I won't have time for a week or so to check -- but can do next week. Ideally, I would like to see regression tests added for the new rules.

**Comment by DavidLRowe** [comment]
@mcdurdin Are you waiting for regression tests to be added before reviewing this?

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Apologies for the long delay in reviewing this. I have carefully checked the decomposed rules and all are correct bar one, I think!

**Comment by MakaraSok** [comment]
I've just got back to this. Even with FLEx 9.0.8.0, the keyboard output the expected characters both in the field and preview pain without any issue.

**Comment by LornaSIL** [comment]
@MakaraSok why was this closed?

**Comment by MakaraSok** [comment]
The issue can no longer be repro-ed.

**Comment by LornaSIL** [comment]
Ok. I hope you weren't using your modified keyboard that isn't in the repo!

**Comment by MakaraSok** [comment]
Details on the test that proves it works as expected:

- Keyman version: 14.0.204-alpha
- Keyboard name: [Galaxie Greek (Mnemonic)](https://keyman.com/keyboards/galaxie_greek_mnemonic) 
- Keyboard version: 3.2.1
- Windows Version: Windows 10 Pro 1909 x64
- FLEx version: 9.0.8.0 RC x64

image

**Pull request closed without merging**